llvm/test/CodeGen/X86/extractelement-fp.ll

```diff
@@ -320,7 +320,8 @@ define <3 x double> @extvselectsetcc_crash(<2 x double> %x) {
 ; X64-LABEL: extvselectsetcc_crash:
 ; X64:       # %bb.0:
 ; X64-NEXT:    vcmpeqpd {{\.?LCPI[0-9]+_[0-9]+}}(%rip), %xmm0, %xmm1
-; X64-NEXT:    vmovsd {{.*#+}} xmm2 = [1.0E+0,0.0E+0]
+; X64-NEXT:    movabsq $4607182418800017408, %rax # imm = 0x3FF0000000000000
```

Comment:
I'm not an expert on X86 assembly, but this test change and the one above seem to produce more instructions so is that a regression?

Reply:
Its an optimization to save a load, I'm not 100% sure if the optimization is sound, but think it should be addressed there not here.

Reply:
vector loads vs rematerialization vs scalar constants has never been ideal on x86